graphql/resolvers/score/mutations.js

```diff
@@ -0,0 +1,200 @@
+
+        let startStand = category.start_stand;
+        let endStand = category.end_stand;
+
+        let pos = startStand;
+        let order = "A".charCodeAt(0);
+
+        for (let score of scores) {
+            score.stand = pos;
+            score.order = String.fromCharCode(order);
+            score.save();
+
+            pos = pos + 1;
+            if (pos > endStand) {
+                pos = startStand;
+                order = order + 1;
+            }
+        }
+
+        return scores;
+    },
+};
+
+function createNewScore(args) {
+    return new Score({
+        ...args,
+        distances: [],
+        pre_elimination_score: 0,
+        access_code: generateCode(),
+    });
+}
+
+async function getDistancesFromCategory(category_id) {
+    let category = await Category.findById(category_id);
+    return category.distances;
+}
+
+function addDistancesToScore(distances, score) {
+    for (let distance of distances) {
+        addDistanceToScore(score, distance);
+    }
+
+    function addDistanceToScore(score, distance) {
+        score.distances.push({
+            distance_id: distance.distance_id,
+            name: distance.name,
+            score: 0,
+        });
+    }
+}
+
+async function createAllSeries(participant_id, category_id, distances) {
+    for (let d of distances) {
+        let distance = await Distance.findById(d.distance_id);
+        for (let i = 0; i < distance.series_type; i++) {
+            await createSeries(distance._id, i);
+        }
+    }
+
+    async function createSeries(distance_id, series_no) {
+        await Series.create({
+            participant_id,
+            category_id,
+            distance_id,
+            series_no,
+            was_counted: false,
+            score: 0,
+            arrows: [],
+            pre_elimination_score: 0,
+        });
+    }
+}
+
+async function updateScoreFromSeries(score, distance_id) {
+    zeroScore();
+
+    let allSeries = await Series.find({
+        distance_id,
+        participant_id: score.participant_id,
+    });
+
+    for (let series of allSeries) {
+        countSeries(series);
+    }
+
+    countPreEliminationScore();
+
+    function zeroScore() {
+        for (let i = 0; i < score.distances.length; i++) {
+            if (score.distances[i].distance_id.toString() === distance_id) {
+                score.distances[i].score = 0;
+            }
+        }
+    }
+
+    function countSeries(series) {
+        for (
+            let distance_idx = 0;
+            distance_idx < score.distances.length;
+            distance_idx++
+        ) {
+            if (
+                score.distances[distance_idx].distance_id.toString() ===
+                distance_id
+            ) {
+                addSeriesToScore(distance_idx, series);
+            }
+        }
+
+        function addSeriesToScore(distance_idx, series) {
+            score.distances[distance_idx].score += series.score;
+            series.was_counted = true;
+        }
```

Comment:
Do you even need classic for loop here? distance_idx is used only as index, so that's shouldn't be the reason.
If score.distances is iterable, use for(let of) or something.